server/routes/groomRoutes.js

```diff
@@ -0,0 +1,87 @@
+const { Router } = require('express');
+const { User, Dog, Groom } = require('../db');
+
+const router = Router();
+
+// Get user's dogs
+router.get('/:userId', (req, res) => {
+  const { userId } = req.params;
+  Dog.find({ owner: userId })
+    // .where()
+    .then((dogsArray) => {
+      User.findById(userId).then(() => {
+        res.status(200).send({ dogsArray });
+      });
+    })
+    .catch((err) => {
+      console.error('SERVER ERROR: failed to GET dog by userId', err);
+      res.sendStatus(500);
+    });
+});
+
+router.get('/', (req, res) => {
+  const { groomId } = req.params;
+  Groom.find({ groom: groomId }).then((groomData) => {
+    res.status(200).send(groomData);
+  });
+});
+
+router.post('/', (req, res) => {
+  const { isSubscribed } = req.body;
+  const status = new Date(new Date().getTime() + 24 * 60 * 60 * 1000);
+
+  Groom.create({
+    isSubscribed,
+    feedDeadline: status,
+    walkDeadline: status,
+  })
+    .then(() => {
+      return User.findByIdAndUpdate(
+        owner,
+        { $inc: { coinCount: -15, dogCount: -1 }, $pull: { breeds: img } },
+        { new: true }
+      ).catch((err) => {
+        console.error('SERVER ERROR: failed to UPDATE user', err);
+        res.sendStatus(500);
+      });
+    })
+    .then((updatedUser) => {
+      res.status(201).send(updatedUser);
+    })
+    .catch((err) => {
+      console.error('SERVER ERROR: failed to CREATE dog', err);
+      res.sendStatus(500);
+    });
+});
+
+router.post('/', (req, res) => {
```

Comment:
Will this conflict with the route starting on line 29?

Reply:
I was testing postman with those, but they will be changed once I have my endpoint working the way I need to. It shouldn't affect anyone elses.